modules/distribution/src/main/scripts/fluo

```diff
@@ -97,60 +124,140 @@ function check_hadoop {
-  copy_config logback.xml
+  init_dir=$(java org.apache.fluo.command.FluoGetProp "$connection_props" "$app_props" fluo.observer.init.dir)
+  if [ -d "$init_dir" ]; then
+    echo "Adding $init_dir/* to CLASSPATH"
+    export CLASSPATH="$init_dir/*:$CLASSPATH"
+  fi
+  java org.apache.fluo.command.FluoSetup "$connection_props" "$app_props" ${*:3}
   ;;
-init)
-  validate_app_full
-  check_hadoop
-  if [[ $@ != *"-h"* && $@ != *"-u"* ]]; then
-    echo "Copying Fluo jars to HDFS at /fluo/lib to be accessible by Accumulo for iterators"
-    $HADOOP_PREFIX/bin/hdfs dfs -mkdir -p /fluo/lib
-    echo "Copying `ls $FLUO_HOME/lib/fluo-api-*.jar` to HDFS"
-    $HADOOP_PREFIX/bin/hdfs dfs -copyFromLocal -f $FLUO_HOME/lib/fluo-api-*.jar /fluo/lib/
-    echo "Copying `ls $FLUO_HOME/lib/fluo-accumulo-*.jar` to HDFS"
-    $HADOOP_PREFIX/bin/hdfs dfs -copyFromLocal -f $FLUO_HOME/lib/fluo-accumulo-*.jar /fluo/lib/
+oracle)
+  setup_service $2
+  nohup java "${JAVA_OPTS[@]}" org.apache.fluo.command.FluoOracle "$connection_props" "$2" > "${FLUO_LOG_DIR}/${FLUO_LOG_ID}.out" 2> "${FLUO_LOG_DIR}/${FLUO_LOG_ID}.err" &
+  echo "$!" > "$basedir/run/${FLUO_LOG_ID}.pid"
+  echo "Started Oracle for '$2' Fluo application. Logs can be found in ${FLUO_LOG_DIR}"
+  ;;
+worker)
+  setup_service $2
+  nohup java "${JAVA_OPTS[@]}" org.apache.fluo.command.FluoWorker "$connection_props" "$2" > "${FLUO_LOG_DIR}/${FLUO_LOG_ID}.out" 2> "${FLUO_LOG_DIR}/${FLUO_LOG_ID}.err" &
+  echo "$!" > "$basedir/run/${FLUO_LOG_ID}.pid"
+  echo "Started Worker for '$2' Fluo application. Logs can be found in ${FLUO_LOG_DIR}"
+  ;;
+scan)
+  if [ -f "$conf/connection.properties" ]; then
+    check_conn_props
+    java org.apache.fluo.command.FluoScan "$connection_props" ${*:2}
+  else 
+    check_hadoop
+    java org.apache.fluo.cluster.command.FluoCommand "$basedir" "$HADOOP_PREFIX" "$@"
   fi
-  export CLASSPATH="$APP_LIB_DIR/*:$CLASSPATH"
-  java org.apache.fluo.cluster.command.FluoCommand $FLUO_HOME $HADOOP_PREFIX "$@"
   ;;
-start)
-  validate_app_full
-  check_hadoop
-  export CLASSPATH="$APP_LIB_DIR/*:$CLASSPATH"
-  java org.apache.fluo.cluster.command.FluoCommand $FLUO_HOME $HADOOP_PREFIX "$@"
+stop)
+  if [ -f "$conf/connection.properties" ]; then
+    echo "Stopping all processes for ${2} application"
+    for pid_file in $basedir/run/${2}_*.pid; do
+      if [ -f "$pid_file" ]; then
+        kill -s KILL "$(cat "$pid_file")" 2>/dev/null
+        rm -f "${pid_file}" 2>/dev/null
+      fi 
+    done
+  else 
+    check_hadoop
+    java org.apache.fluo.cluster.command.FluoCommand "$basedir" "$HADOOP_PREFIX" "$@"
+  fi
   ;;
-stop|kill|status|info|scan|wait)
-  validate_app
-  check_hadoop
-  java org.apache.fluo.cluster.command.FluoCommand $FLUO_HOME $HADOOP_PREFIX "$@"
+ps) 
+  jps -m | grep Fluo
   ;;
 list)
-  check_hadoop
-  java org.apache.fluo.cluster.command.FluoCommand $FLUO_HOME $HADOOP_PREFIX $1 app ${*:2}
+  if [ -f "$conf/connection.properties" ]; then
```

Comment:
Instead of checking for non-existence, could checks that connection.properties has no props set.  This would allow old code that only modifies fluo.props to work w/o having to delete this file.